forge-game/src/main/java/forge/game/phase/PhaseHandler.java

```diff
@@ -468,6 +472,9 @@ private void onPhaseEnd() {
             throw new IllegalStateException("Phase.nextPhase() is called, but Stack isn't empty.");
         }
 
+        // now that we're going, give priority to the player
+        givePriorityToPlayer = true;
```

Comment:
Good catch. It looks like we used to do it at the end of the phase because we had to do it somewhere. Right now we set it to false in onPhaseBegin() if we want to skip priority, but looking at the comprehensive rules it looks like we really should set it each time in there with references to the rule that says we should have it or not.

Reply:
Reading through the comprehensive rules and our current implementation, it looks like we should be fully skipping some parts of combat phases, instead of just not giving the player priority.